src/nfts/vendors/puzzle.ts

```diff
@@ -0,0 +1,81 @@
+import { DataTransactionEntryString } from '@waves/ts-types/src/parts';
+
+import {
+  dataEntriesToRecord,
+  fetchDataEntries,
+} from '../../nodeApi/dataEntries';
+import {
+  CreateParams,
+  FetchInfoParams,
+  Nft,
+  NftAssetDetail,
+  NftVendor,
+  NftVendorId,
+} from '../types';
+
+interface PuzzleNftInfo {
+  id: string;
+  vendor: NftVendorId.Puzzle;
+  image: string;
+  creator: string;
+}
+
+const PUZZLE_MARKET_DAPP = '3PFQjjDMiZKQZdu5JqTHD7HwgSXyp9Rw9By';
+
+function puzzleDataUrl(nodeUrl: string) {
+  return new URL(`addresses/data/${PUZZLE_MARKET_DAPP}`, nodeUrl).toString();
+}
+
+function nftIdKey(id: string, postfix = '') {
+  return `nft_${id}${postfix}`;
+}
```

Comment:
Maybe include `_` from `postfix` in string format and rename the function to `nftPropertyKey`?